naver-map-compose/src/main/java/com/naver/maps/map/compose/NaverMap.kt

```diff
@@ -177,6 +179,15 @@ private fun MapLifecycle(mapView: MapView) {
         onDispose {
             mapView.onSaveInstanceState(savedInstanceState)
             lifecycle.removeObserver(mapLifecycleObserver)
+            if (mapLifecycleObserver.receivedEvents.contains(Lifecycle.Event.ON_PAUSE).not()) {
+                mapView.onPause()
+            }
+            if (mapLifecycleObserver.receivedEvents.contains(Lifecycle.Event.ON_STOP).not()) {
+                mapView.onStop()
+            }
+            if (mapLifecycleObserver.receivedEvents.contains(Lifecycle.Event.ON_DESTROY).not()) {
+                mapView.onDestroy()
+            }
             context.unregisterComponentCallbacks(callbacks)
```

Comment:
(1) 예외처리로 들어가는 부분이니, callback 해제하는 부분 아래로 옮겨주시구요.
(2) Set 대신 마지막 상태에 따라 처리 부탁드립니다.
(3) 주석도 적절히 추가 부탁드려요.

```suggestion
            context.unregisterComponentCallbacks(callbacks)
            
            // workaround:
            // dispose 시점에 Lifecycle.Event가 끝까지 진행되지 않아 발생되는
            // MapView Memory Leak 수정합니다.
            when (previousState.value) {
                Lifecycle.Event.ON_CREATE -> {
                    mapView.onDestroy()
                }
                Lifecycle.Event.ON_START -> {
                    mapView.onStop()
                    mapView.onDestroy()
                }
                Lifecycle.Event.ON_RESUME -> {
                    mapView.onPause()
                    mapView.onStop()
                    mapView.onDestroy()
                }
                else -> {}
            }
```